should not panic when load with invalid dump file

**What happened**:
```
root@bench-01:~/juicefs# cat test.json
{"a": 1}
root@bench-01:~/juicefs# rm test.db
root@bench-01:~/juicefs# ./juicefs load sqlite3://test.db test.json
2024/04/23 12:14:47.338041 juicefs[602757] <INFO>: Meta address: sqlite3://test.db [interface.go:504]
2024/04/23 12:14:47.409179 juicefs[602757] <INFO>: Loading from file ... [dump.go:377]
Loaded entries: 0/0 [--------------------------------------------------------------]  0.0/s used: 326.805µs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1444c7f]

goroutine 1 [running]:
github.com/juicedata/juicefs/pkg/meta.loadEntries({0x3dcefa0?, 0xc00083e458}, 0x3158720?, 0x0?)
	/root/juicefs/pkg/meta/dump.go:426 +0x8bf
github.com/juicedata/juicefs/pkg/meta.(*dbMeta).LoadMeta(0xc0010f4b40, {0x3dcefa0, 0xc00083e458})
	/root/juicefs/pkg/meta/sql.go:3945 +0xce9
github.com/juicedata/juicefs/cmd.load(0xc0016b6000)
	/root/juicefs/cmd/load.go:133 +0xa89
github.com/urfave/cli/v2.(*Command).Run(0xc001684ea0, 0xc000fcba40)
	/root/go/pkg/mod/github.com/juicedata/cli/v2@v2.19.4-0.20230605075551-9c9c5c0dce83/command.go:177 +0x742
github.com/urfave/cli/v2.(*App).RunContext(0xc000fd2380, {0x3dea5d8?, 0xc0000560b0}, {0xc000fca780, 0x4, 0x4})
	/root/go/pkg/mod/github.com/juicedata/cli/v2@v2.19.4-0.20230605075551-9c9c5c0dce83/app.go:389 +0x10e6
github.com/urfave/cli/v2.(*App).Run(...)
	/root/go/pkg/mod/github.com/juicedata/cli/v2@v2.19.4-0.20230605075551-9c9c5c0dce83/app.go:254
github.com/juicedata/juicefs/cmd.Main({0xc000fca040?, 0x4?, 0x4?})
	/root/juicefs/cmd/main.go:101 +0x3aec
main.main()
	/root/juicefs/main.go:29 +0x2e
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/dump.go
+++ b/pkg/meta/dump.go
@@ -382,7 +382,9 @@ func loadEntries(r io.Reader, load func(*DumpedEntry), addChunk func(*chunkKey))
 
 	progress := utils.NewProgress(false)
 	bar := progress.AddCountBar("Loaded entries", 1) // with root
-	dm = &DumpedMeta{}
+	dm = &DumpedMeta{
+		Counters: &DumpedCounters{},
+	}
 	counters = &DumpedCounters{ // rebuild counters
 		NextInode: 2,
 		NextChunk: 1,

--- a/pkg/meta/dump.go
+++ b/pkg/meta/dump.go
@@ -302,7 +302,7 @@ type DumpedMeta struct {
 
 func (dm *DumpedMeta) validate() error {
 	if dm.Counters == nil {
-		return errors.New("invalid dumped meta: missing counters")
+		return errors.New("invalid dumped meta: missing 'Counters'")
 	}
 	return nil
 }

--- a/pkg/meta/dump.go
+++ b/pkg/meta/dump.go
@@ -18,6 +18,7 @@ package meta
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
 	"io"
 	"strings"
@@ -299,6 +300,13 @@ type DumpedMeta struct {
 	Trash     *DumpedEntry         `json:",omitempty"`
 }
 
+func (dm *DumpedMeta) validate() error {
+	if dm.Counters == nil {
+		return errors.New("invalid dumped meta: missing counters")
+	}
+	return nil
+}
+
 func (dm *DumpedMeta) writeJsonWithOutTree(w io.Writer) (*bufio.Writer, error) {
 	if dm.FSTree != nil || dm.Trash != nil {
 		return nil, fmt.Errorf("invalid dumped meta")
@@ -382,9 +390,7 @@ func loadEntries(r io.Reader, load func(*DumpedEntry), addChunk func(*chunkKey))
 
 	progress := utils.NewProgress(false)
 	bar := progress.AddCountBar("Loaded entries", 1) // with root
-	dm = &DumpedMeta{
-		Counters: &DumpedCounters{},
-	}
+	dm = &DumpedMeta{}
 	counters = &DumpedCounters{ // rebuild counters
 		NextInode: 2,
 		NextChunk: 1,
@@ -425,6 +431,11 @@ func loadEntries(r io.Reader, load func(*DumpedEntry), addChunk func(*chunkKey))
 	}
 	_, _ = dec.Token() // }
 	progress.Done()
+
+	if err = dm.validate(); err != nil {
+		return
+	}
+
 	logger.Infof("Dumped counters: %+v", *dm.Counters)
 	logger.Infof("Loaded counters: %+v", *counters)
 	return

